jobs/garbage-collection.Jenkinsfile

```diff
@@ -0,0 +1,89 @@
+node {
+    checkout scm
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+    pipecfg = pipeutils.load_pipecfg()
+    gc_policy_data = pipeutils.load_gc()
+    repo = "coreos/fedora-coreos-config"
+}
+
+// This job runs the Garbage collection on the selected stream in accordance with the
+// gc-policy.yaml available in https://github.com/coreos/fedora-coreos-pipeline. If
+// the pruning step succeeds, it uploads the updated builds/builds.json to s3.
+properties([
+    pipelineTriggers([]),
+    parameters([
+        choice(name: 'STREAM',
+               choices: pipeutils.get_streams_choices(pipecfg),
+               description: 'CoreOS stream to run GC'),
+        booleanParam(name: 'DRY_RUN',
+                     defaultValue: false,
+                     description: 'Dry run Garbage Collection')
+    ])
+])
+
+def build_description = "[${params.STREAM}]"
+def cosa_img = 'quay.io/coreos-assembler/coreos-assembler:main'
+def container_env = pipeutils.get_env_vars_for_stream(pipecfg, params.STREAM)
+def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+def dry_run = params.DRY_RUN ? "--dry-run" : ""
+
+lock(resource: "gc-${params.STREAM}") {
+    cosaPod(image: cosa_img, env: container_env,
+            memory: "1024Mi",
+            serviceAccount: "jenkins") {
+        try {
+            currentBuild.description = "${build_description} Running"
+
+            stage('Init') {
+                def branch = params.STREAM
+                sh "cosa init --branch \"${branch}\" https://github.com/${repo}"
+            }
+
+            // Write YAML data to a new file in cosaPod
+            def new_gc_policy_path = './gc-policy.yaml'
+            writeYaml file: new_gc_policy_path, data: gc_policy_data
+
+            stage('BuildFetch') {
+                pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa buildfetch --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                    --stream ${params.STREAM} --url=s3://${s3_stream_dir}/builds
+                """)
+            }
+
+            withCredentials([file(variable: 'GCP_KOLA_TESTS_CONFIG', credentialsId: 'gcp-image-upload-config')]) {
+                def acl = pipecfg.s3.acl ?: 'public-read'
+
+                stage('Garbage Collection') {
+                    pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                    cosa cloud-prune --policy ${new_gc_policy_path} \
+                    --stream ${params.STREAM} ${dry_run} \
+                    --gcp-json-key=\${GCP_KOLA_TESTS_CONFIG} \
+                    --acl=${acl} \
+                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
+                    """)
+                }
+            }
+
+            // Nested lock for the Upload Builds JSON step
+            lock(resource: "builds-json-${params.STREAM}") {
```

Comment:
We should just make this whole block conditional on whether it's a no-op. That way e.g. we don't lock, or show a misleading stage in the BlueOcean UI.

You can read the timestamp from builds.json before calling `cosa cloud-prune` and then compare the timestamp after.